### ISSUE TITLE
autolabeler needs option to avoid labeling already labeled issues

### ISSUE THREAD

**Issue by ssbarnea** [release-drafter/release-drafter#1114]
While autolabeler is a wonderful tool for adding missing labels to pull requests coming from external contributors it does also have a dark side, especially if you dare to add patterns for title or branch name.

If a PR was created from a branch like `chore/foo` and you have a pattern to make as `skip-changelog`, even if you decide that it should be counted as a `bug` and remove the wrong label, the tool will add it back (if you configured it to run on edits, which you are likely to).

Basically autolabeler runs and overrides manually configured labels and there is not way to configured it to do nothing if someone manually labeled a pull request.

This downside basically prevents us from making use of pattern matches.

The solution for this should be to add an option to make changes only on unlabeled pull requests. We can assume that a new pull request will have no labels and that anyone with triage rights knows what they are doing, so there is no need to override human decision.

Here is such an example: https://github.com/ansible/schemas/pull/228

**Comment by ssbarnea** [comment]
Can we get at least an ack from the maintainers that they would welcome a pull request adding this ability?

We love the ability to guess the labels based on branch name or title but when this goes wrong and the maintainer is fixing it manually... it is reverted by the next run of the labeler. Basically we do not want to relabel once labels were added by a human (or at least once a set of labels are found to already present).

**Comment by jetersen** [comment]
I'll we willing and able to go through all issues once the major release is done.

I am planning to move autolabeler to it's own action where it will be easier to maintain.

**Comment by ssbarnea** [comment]
I am afraid that we need to ask, I know several open-source projects where the maintainer have... little interest in getting features from the users and they are not really open about that. Spending hours or days on a PR just to discover that maintainer refuses it is very annoying. That is why I prefer to ask. Also as maintainer I can say that I was pleasantly surprised about getting a pull request with the fix after mentioning that I would welcome it. That is why I recently decided to add template, one that I use especially for tickets which I agree with but where I doubt I would personally have time to implement them myself.

**Comment by ssbarnea** [comment]
Did anyone switch to use of another labeling action in order to overcome the incomplete implementation embedded into release drafter?

**Comment by cchanche** [comment]
I think we should address this by implementing a new filter, that selects whether a `pull_request` target applies for autolabeler (at all).

Using Github Actions syntax, you can already do this :

```yaml
on:
  pull_request:
    types: [opened, reopened, synchronize]

permissions:
  contents: read

jobs:
  autolabeler:
    name: 'Autolabeler'
    permissions:
      pull-requests: write
    runs-on: ubuntu-latest
    if: !github.event.pull_request.labels[0] # falsy of pr has at least one label
    steps:
      - uses: release-drafter/release-drafter/autolabeler@v7
```

I am unsure it's wise to reimplement such a thing locally (in autolabeler's code).

If we were to solve your issue in a more deterministic way, I believe we'd have to find a way to find out about managed vs unmanaged labels (which has been discussed in #1432 for similar concerns). It is kind of hard to nail.

Considering the Github Actions expression syntax workaround, and the age of this issue, I am closing this. Feel free to reopen if needed.
